### DISCUSSION TITLE
RoderickShao_notebook/%E8%AE%A1%E7%AE%97%E6%9C%BA%E5%9F%BA%E7%A1%80%E8%AF%BE/%E9%AB%98%E7%BA%A7%E6%95%B0%E6%8D%AE%E7%BB%93%E6%9E%84%E4%B8%8E%E7%AE%97%E6%B3%95%E5%88%86%E6%9E%90/Lecture%2012/

### DISCUSSION THREAD

**Discussion by giscus** [RoderickShao/RoderickShao_notebook#8] [Announcements]
# RoderickShao_notebook/%E8%AE%A1%E7%AE%97%E6%9C%BA%E5%9F%BA%E7%A1%80%E8%AF%BE/%E9%AB%98%E7%BA%A7%E6%95%B0%E6%8D%AE%E7%BB%93%E6%9E%84%E4%B8%8E%E7%AE%97%E6%B3%95%E5%88%86%E6%9E%90/Lecture%2012/

RoderickShao's NoteBook

https://roderickshao.github.io/RoderickShao_notebook/%E8%AE%A1%E7%AE%97%E6%9C%BA%E5%9F%BA%E7%A1%80%E8%AF%BE/%E9%AB%98%E7%BA%A7%E6%95%B0%E6%8D%AE%E7%BB%93%E6%9E%84%E4%B8%8E%E7%AE%97%E6%B3%95%E5%88%86%E6%9E%90/Lecture%2012/

**Comment by Provence-24** [comment]
我对K-L算法的原理有一个猜测。因为它的过程里面包含了往相对差的情况走，这有点像Metropolis算法中有一定概率往坏的情况走。这样不会和每次走一步的相邻搜素方式一样，很容易陷入一个local optimal。它更有可能走出这个小局部，去其他地方探索。所以K-L允许某几步往更差的方向走，而基础的方法不允许这样的事情发生，一旦只能更差，那么直接不走结束了。这也就是扩大邻居范围的好处

**Reply by RoderickShao** [reply]
对的，以合理的方式扩大邻居的范围
